packages/flutter/lib/src/painting/border_radius.dart

```diff
@@ -26,6 +26,82 @@ abstract class BorderRadiusGeometry {
   /// const constructors so that they can be used in const expressions.
   const BorderRadiusGeometry();
 
+  /// Creates a [BorderRadius] where all radii are `radius`.
+  // The radius applies equally on all sides, so BorderRadiusDirectional is
+  // irrelevant in this case.
+  factory BorderRadiusGeometry.all(Radius radius) => BorderRadius.all(radius);
+
+  /// Creates a [BorderRadius] where all radii are [Radius.circular(radius)].
+  // The radius applies equally on all sides, so BorderRadiusDirectional is
+  // irrelevant in this case.
+  factory BorderRadiusGeometry.circular(double radius) => BorderRadius.circular(radius);
+
+  /// Creates a horizontally symmetrical border radius.
+  ///
+  /// Utilizing the `left` and `right` properties will return a [BorderRadius],
+  /// while `start` and `end` will yield a [BorderRadiusDirectional]. These
+  /// properties cannot be used interchangeably.
+  factory BorderRadiusGeometry.horizontal({
+    Radius? left, // BorderRadius
+    Radius? right, // BorderRadius
+    Radius? start, // BorderRadiusDirectional
+    Radius? end, // BorderRadiusDirectional
+  }) {
+    assert(
+      (left == null && right == null) || (start == null && end == null),
+      'The left and right values cannot be used in conjunction with start and end.',
+    );
```

Comment:
If everything is null, it will create a BorderRadius.horizontal with both parameters set to Radius.zero. I think that's what we want, but just wanted to call that out to double check.

Reply:
Indeed! Thanks!